main/src/cgeo/geocaching/models/CalcState.java

```diff
@@ -0,0 +1,135 @@
+package cgeo.geocaching.models;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+
+import cgeo.geocaching.settings.Settings;
+import cgeo.geocaching.ui.CalculateButton;
+import cgeo.geocaching.ui.CalculatorVariable;
+
+/**
+ * This class is designed to capture the current state of the coordinate calculator such that it can be preserver for latter use.
+ **
+ * All the relevant information is in a serializable form such that it can be stored as a bundle in waypoint's 'ContentValues'.
+ **/
+
+public class CalcState implements Serializable {
+    public static final char ERROR_CHAR = '#';
+    public static final String ERROR_STRING = "???";
+
+    public Settings.CoordInputFormatEnum format;
+    public String plainLat, plainLon;
+    public char latHemisphere, lonHemisphere;
+    public List<CalculateButton.ButtonData> buttons;
+    public List<CalculatorVariable.VariableData> equations, freeVariables;
+    public String notes; // Note, we have to use a String rather than an Editable as Editable's can't be serialized
+
+    public CalcState(final Settings.CoordInputFormatEnum format,
+                     final String plainLat,
+                     final String plainLon,
+                     final char latHem,
+                     final char lonHem,
+                     final List<CalculateButton.ButtonData> buttons,
+                     final List<CalculatorVariable.VariableData> equations,
+                     final List<CalculatorVariable.VariableData> freeVariables,
+                     final String notes) {
+        this.format = format;
+        this.plainLat = plainLat;
+        this.plainLon = plainLon;
+        latHemisphere = latHem;
+        lonHemisphere = lonHem;
+        this.buttons = buttons;
+        this.equations = equations;
+        this.freeVariables = freeVariables;
+        this.notes = notes;
+    }
+
+    public CalcState(final JSONObject json) {
+        format = Settings.CoordInputFormatEnum.values()[json.optInt("format", 2)];
+        plainLat = json.optString("plainLat");
+        plainLon = json.optString("plainLon");
+        latHemisphere = (char) json.optInt("latHemisphere", ERROR_CHAR);
+        lonHemisphere = (char) json.optInt("lonHemisphere", ERROR_CHAR);
+
+        {
```

Comment:
Okay, after yet another evening of hacking I've managed to find a way to implement what I understand to be a "factory" paradigm to get around the blessid 'constructor in the loop' issue.  Hopefully this attempt will make the grade ;).

This is my first attempt at using a technique like this so I'll be more than happy to neaten this up more if anyone can suggest anything that might help ;).